### PR TITLE
fix(nuxt): use `@nuxt/kit`

### PR DESCRIPTION
**Pull request by pi0**
Hi! The latest kit version is now available as `@nuxt/kit`. See [nuxt3-stubs](https://github.com/nuxt/nuxt3-stubs). This will help avoid alias but also reference to non `-edge` version.

I've also replaced `.cjs` build with a proxy since `@nuxt/kit` is an ESM-only package and updated `index` with `module` that will be eventually convention in new module templates.

### CODE REVIEW

File: packages/nuxt/package.json [hunk 1]
```diff
@@ -26,16 +26,16 @@
   "sideEffects": false,
   "exports": {
     ".": {
-      "require": "./dist/index.js",
-      "import": "./dist/index.mjs"
+      "import": "./dist/module.mjs",
```
Comment (posva) [inline packages/nuxt/package.json:29]: shouldn't this be renamed to `./dist/index.cjs`? that way siroc will generate the file

File: packages/nuxt/module.cjs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const kit = require('@nuxt/kit');
+const vueDemi = require('vue-demi');
+
+const module$1 = kit.defineNuxtModule({
+  name: "pinia",
+  configKey: "pinia",
+  defaults: {
+    disableVuex: true
```
Comment (posva) [inline packages/nuxt/module.cjs]: I think this file was included by accident

File: packages/nuxt/package.json [hunk 3]
```diff
@@ -26,16 +26,16 @@
...
+      "require": "./module.cjs"
```
Comment (posva) [inline packages/nuxt/package.json:30]: shouldn't this be renamed to `./dist/index.cjs`? that way siroc will generate the file
```suggestion
      "require": "./dist/index.cjs"
```

File: packages/nuxt/module.cjs (hunk 2 above)
Comment (pi0) [reply packages/nuxt/module.cjs]: Ah I mistakenly replaced it on the last commit.

File: packages/nuxt/package.json (hunk 3 above)
Comment (pi0) [reply packages/nuxt/package.json:30]: This is a manual file and only for legacy nuxt support. Rollup's generated cjs format with require syntax (`require('@nuxt/kit')`) is invalid.

Comment (posva) [reply packages/nuxt/package.json:30]: I see, thanks!

**Review by posva — APPROVED** [review]
Thanks!

**Pull request merged**